File: EasyFinance.DataAccess/Entities/Receipt.cs
using System;
using EasyFinance.DataAccess.Identity;

namespace EasyFinance.DataAccess.Entities
{
    public class Receipt
    {
        public int Id { get; set; }

        public int UserId { get; set; }

        public User User { get; set; }

        public string Merchant { get; set; }

        public string Description { get; set; }

        public int? ReceiptPhotoId { get; set; }

        public ReceiptPhoto ReceiptPhoto { get; set; }

        public int? CategoryId { get; set; }

        public Category Category { get; set; }

        public int? PaymentMethodId { get; set; }

        public PaymentMethod PaymentMethod { get; set; }

        public decimal? TotalAmount { get; set; }

        public int? CurrencyId { get; set; }

        public Currency Currency { get; set; }

        public DateTime? PurchaseDate { get; set; }
    }
}

File: EasyFinanceUI/src/app/models/receipt.ts
import { Category } from './category';
import { PaymentMethod } from './payment-method';
import { Currency } from './currency';
import { ReceiptPhoto } from './receipt-photo';

export class Receipt {
    public id: number;
    public merchant: string;
    public receiptPhotoId: number;
    public receiptPhoto: ReceiptPhoto;
    public categoryId: number;
    public category: Category;
    public paymentMethodId: number;
    public paymentMethod: PaymentMethod;
    public totalAmount: number;
    public currencyId: number;
    public currency: Currency;
    public purchaseDate: Date;
    public description: string;


    constructor() {
        this.id = 0;
        this.merchant = '';
        this.receiptPhotoId = null;
        this.categoryId = null;
        this.paymentMethodId = null;
        this.totalAmount = 0.00;
        this.currencyId = null;
        this.purchaseDate = new Date();
        this.description = ''
    }
}
